### PR TITLE
CorsixTH

### PR DESCRIPTION
**Pull request by alexandre-lavoie**
###### Motivation for this change

This is a Nix port for the [CorsixTH](https://github.com/CorsixTH/CorsixTH). This has been developed in https://github.com/CorsixTH/CorsixTH/pull/1881 and accepted by maintainers on CorsixTH branch. This PR makes minor changes to fit Nixpkgs guidelines. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/corsix-th/level-edit.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    '';
+
+    installPhase = ''
+        mkdir -p $out/bin $out/share/java
+
+        cp dist/LevelEdit.jar $out/share/java
+
+        makeWrapper ${openjdk11}/bin/java $out/bin/${pname} \
+            --add-flags "-jar $out/share/java/LevelEdit.jar" \
+            --prefix PATH : "$wrappedPath"
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/level-edit.nix:29]:
```suggestion
          --add-flags "-jar $out/share/java/LevelEdit.jar" \
          --prefix PATH : "$wrappedPath"
```

File: pkgs/games/corsix-th/level-edit.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ baseName ? "corsixth", version ? "trunk", stdenv, lib, githubSource, openjdk11, makeWrapper, ant }:
+
+with lib;
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/level-edit.nix:5]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/games/corsix-th/default.nix [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    version = "0.64";
+    githubSource = fetchFromGitHub {
+        owner = "CorsixTH";
+        repo = "CorsixTH";
+        rev = "8741d41eb5de865c633e023bcb512b779c131c5e";
+        sha256 = "0chh9cv2kdc39sr0x8hclcyzd8dz2y6grgagqzkvr7j570wa5cqh";
+    };
+in
+lib.makeScope newScope (self: with self; {
+    corsix-th = callPackage ./corsix-th.nix { inherit baseName; inherit version; inherit githubSource; inherit audioSupport; inherit movieSupport; inherit freetypeSupport; inherit buildDocs; inherit enableUnitTests; };
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/default.nix:14]:
```suggestion
    corsix-th = callPackage ./corsix-th.nix { inherit baseName version githubSource audioSupport; movieSupport freetypeSupport buildDocs enableUnitTests; };
```

File: pkgs/games/corsix-th/default.nix [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    # anim-view = if buildAnimView then callPackage ./anim-view.nix { inherit baseName; inherit version; inherit githubSource; } else null;
+    level-edit = if buildLevelEdit then callPackage ./level-edit.nix { inherit baseName; inherit version; inherit githubSource; } else null;
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/default.nix:16]:
```suggestion
    level-edit = callPackage ./level-edit.nix { inherit baseName version githubSource; };
```
packages are not allowed to be null in most cases.

File: pkgs/games/corsix-th/default.nix [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{ fetchFromGitHub, lib, newScope, audioSupport ? true, movieSupport ? true, freetypeSupport ? true, buildDocs ? false, enableUnitTests ? false, buildAnimView ? true, buildLevelEdit ? true }:
+
+let
+    baseName = "corsix-th";
+    version = "0.64";
+    githubSource = fetchFromGitHub {
+        owner = "CorsixTH";
+        repo = "CorsixTH";
+        rev = "8741d41eb5de865c633e023bcb512b779c131c5e";
+        sha256 = "0chh9cv2kdc39sr0x8hclcyzd8dz2y6grgagqzkvr7j570wa5cqh";
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/default.nix:10]:
```suggestion
      owner = "CorsixTH";
      repo = "CorsixTH";
      rev = "v%{version}";
      sha256 = "0chh9cv2kdc39sr0x8hclcyzd8dz2y6grgagqzkvr7j570wa5cqh";
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+    preFixup = ''
+        cp -a "${lua.packageDir}"/. $out/share/corsix-th/
+    '';
+
+    postInstall = ''
+        mkdir -p $out/share/bin
+        mv $out/bin/${baseName} $out/share/bin
+        makeWrapper $out/share/bin/${baseName} $out/bin/${baseName} \
+            --set SDL_SOUNDFONTS ${soundfont-fluid.outPath}/share/soundfonts/FluidR3_GM2-2.sf2
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:69]:
```suggestion
          --set SDL_SOUNDFONTS ${soundfont-fluid.outPath}/share/soundfonts/FluidR3_GM2-2.sf2
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+        "-DFFMPEG_DIR=${optionalString movieSupport ffmpeg.outPath}"
+        "-DFREETYPE_DIR=${optionalString freetypeSupport freetype.outPath}"
+        "-DSDL_MIXER_DIR=${optionalString audioSupport SDL2_mixer.outPath}"
+    ]
+        ++ optional (!audioSupport) "-DWITH_AUDIO=OFF"
+        ++ optional (!freetypeSupport) "-DWITH_FREETYPE2=OFF"
+        ++ optional (!movieSupport) "-DWITH_MOVIES=OFF"
+        ++ optional enableUnitTests "-DENABLE_UNIT_TESTS=ON"
+    ;
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:59]:
```suggestion
    ] ++ optional (!audioSupport) "-DWITH_AUDIO=OFF"
      ++ optional (!freetypeSupport) "-DWITH_FREETYPE2=OFF"
      ++ optional (!movieSupport) "-DWITH_MOVIES=OFF"
      ++ optional enableUnitTests "-DENABLE_UNIT_TESTS=ON";
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        ++ optional movieSupport ffmpeg
+        ++ optional buildDocs doxygen
+        ++ optional enableUnitTests catch2
+    ;
+
+    cmakeFlags = [
+        "-DLUA_DIR=${lua.env.outPath}"
+        "-DLUA_PACKAGES_DIR=${lua.packageDir}"
+        "-DSDL_LIBRARY=${SDL2.outPath}/lib/libSDL2.so"
+        "-DSDL_INCLUDE_DIR=${SDL2.dev.outPath}/include/SDL2"
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:49]:
```suggestion
        "-DLUA_DIR=${lua.env}"
        "-DLUA_PACKAGES_DIR=${lua.packageDir}"
        "-DSDL_LIBRARY=${SDL2}/lib/libSDL2.so"
        "-DSDL_INCLUDE_DIR=${SDL2.dev}/include/SDL2"
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        lua.env
+        SDL2
+    ]
+        ++ optional audioSupport SDL2_mixer_fluid
+        ++ optionals (audioSupport && stdenv.isLinux) [ soundfont-fluid fluidsynth ]
+        ++ optional freetypeSupport freetype
+        ++ optional movieSupport ffmpeg
+        ++ optional buildDocs doxygen
+        ++ optional enableUnitTests catch2
+    ;
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:43]:
```suggestion
    ] ++ optional audioSupport SDL2_mixer_fluid
      ++ optionals (audioSupport && stdenv.isLinux) [ soundfont-fluid fluidsynth ]
      ++ optional freetypeSupport freetype
      ++ optional movieSupport ffmpeg
      ++ optional buildDocs doxygen
      ++ optional enableUnitTests catch2;
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        configureFlags = old.configureFlags ++ [
+            " --enable-music-midi-fluidsynth-shared"
+            " --disable-music-midi-timidity"
+        ];
+    }) else null;
+in
+stdenv.mkDerivation rec {
+    inherit version;
+
+    pname = "${baseName}";
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:24]:
```suggestion
    pname = baseName;
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+, SDL2_mixer ? null, soundfont-fluid ? null, fluidsynth ? null, ffmpeg ? null, freetype ? null, doxygen ? null, catch2 ? null
+, audioSupport ? true, movieSupport ? true, freetypeSupport ? true, buildDocs ? false, enableUnitTests ? false
+}:
+
+with lib;
+
+let
+    lua = {
+        env = lua5_3.withPackages(ps: with ps; [ lpeg luafilesystem ]);
+        packageDir = "${lua.env.outPath}/lib/lua/5.3";
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:11]:
```suggestion
        packageDir = "${lua.env}/lib/lua/5.3";
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 12]
```diff
@@ -0,0 +1,80 @@
+{ baseName ? "corsix-th", version ? "trunk", stdenv, cmake, lib, githubSource, SDL2, lua5_3, makeWrapper
+, SDL2_mixer ? null, soundfont-fluid ? null, fluidsynth ? null, ffmpeg ? null, freetype ? null, doxygen ? null, catch2 ? null
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:2]:
```suggestion
, SDL2_mixer, soundfont-fluid, fluidsynth, ffmpeg, freetype, doxygen, catch2
```

File: pkgs/games/corsix-th/anim-view.nix [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    makeFlags = [
+        "AnimView"
+    ];
+
+    postInstall = ''
+        mkdir -p $out/bin
+
+        cp $out/AnimView/AnimView $out/bin/${pname}
+
+        rm -rf $out/AnimView
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/anim-view.nix:32]:
```suggestion
        rm -r $out/AnimView
```

File: pkgs/games/corsix-th/anim-view.nix [hunk 14]
```diff
@@ -0,0 +1,43 @@
+{ baseName ? "corsix-th", version ? "trunk", stdenv, lib, githubSource, cmake, wxGTK30 }:
+
+with lib;
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/anim-view.nix:5]:
```suggestion
stdenv.mkDerivation rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please format the new nix-files with nixpkgs-fmt.

File: pkgs/games/corsix-th/default.nix (hunk 5 above)
Comment (alexandre-lavoie) [reply pkgs/games/corsix-th/default.nix:10]: This will have to be kept commit specific until 0.65 (which should be released soon!)

**Comment by alexandre-lavoie** [comment]
Thank you for the thorough review. I've applied the manual changes + the nixpkgs formatting. Let me know if there are additional changes that you noticed!

Have a good day.

**Comment by alexandre-lavoie** [comment]
Version has been bumped to latest release. Main game and additional programs are working as intended.

Let me know if there is additional changes required for merging.

File: pkgs/games/corsix-th/level-edit.nix [hunk 15]
```diff
@@ -0,0 +1,38 @@
+{ baseName ? "corsixth", version ? "trunk", stdenv, lib, githubSource, openjdk11, makeWrapper, ant }:
+
+stdenv.mkDerivation rec {
+  inherit version;
+
+  pname = "${baseName}-leveledit";
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/level-edit.nix:6]:
```suggestion
  pname = "${baseName}-leveledit";
  inherit version;
```

File: pkgs/games/corsix-th/corsix-th.nix [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+
+  SDL2_mixer_fluid =
+    if audioSupport then
+      SDL2_mixer.overrideAttrs
+        (old: rec {
+          configureFlags = old.configureFlags ++ [
+            " --enable-music-midi-fluidsynth-shared"
+            " --disable-music-midi-timidity"
+          ];
+        }) else null;
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/corsix-th.nix:19]:
```suggestion
  SDL2_mixer_fluid = SDL2_mixer.overrideAttrs
    (old: rec {
      configureFlags = old.configureFlags ++ [
        " --enable-music-midi-fluidsynth-shared"
        " --disable-music-midi-timidity"
      ];
    });
```
This is not evaled when disabled.

File: pkgs/games/corsix-th/anim-view.nix [hunk 17]
```diff
@@ -0,0 +1,37 @@
+{ baseName ? "corsix-th", version ? "trunk", stdenv, lib, githubSource, cmake, wxGTK30 }:
+
+stdenv.mkDerivation rec {
+  inherit version;
+
+  pname = "${baseName}-animview";
```
Comment (SuperSandro2000) [inline pkgs/games/corsix-th/anim-view.nix:6]:
```suggestion
  pname = "${baseName}-animview";
  inherit version;
```

**Comment by revuwa** [comment]
Seems, something is going on in other PRs:

- https://github.com/NixOS/nixpkgs/pull/241203
- https://github.com/NixOS/nixpkgs/pull/250674

`nix run nixpkgs#corsix-th`
already works for me under nixos-unstable, ***yay!***

**Comment by AndersonTorres** [comment]
Closing as dead and solved.

**Pull request closed without merging**